EZP-32219: Fix ContentType sorting defaults has no effect on new created objects

ITernovtsiy opened the pull request:
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | EZP-32219
| **Type**                                   | bug
| **Target eZ Platform version** | v3.2
| **BC breaks**                          | no
| **Doc needed**                       | yes

**Steps to reproduce:**
1. Create new ContentType i.e. "News".
2. Add a single Textline field i.e. "Title".
3. Set "Sort children by default by" to "Location priority"
4. Set "Sort children by default in order" to "Descending".
5. Create new News content anywhere.
6. Observe Sub-items sorting order in the Details tab for this content.

**Expected result:**
Sub-items sorting for new content should have the same value as one that has been set in ContentType.

**Actual result:**
Sub-items sorting for new content is not the same as one that has been set in this ContentType.

#### Checklist:
- [x] Provided PR description.
- [x] Tested the solution manually.
- [x] Provided automated test coverage.
- [x] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [x] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [x] Asked for a review (ping `@ezsystems/php-dev-team`).

ITernovtsiy commented:
ping @ezsystems/php-dev-team

ITernovtsiy commented:
failed tests and code duplications are unrelated to the PR

adamwojs commented:
> failed tests and code duplications are unrelated to the PR

Confirm. Unit tests are failing on 1.0 branch as well.

adamwojs commented:
@ITernovtsiy Unit tests has been fixed in #140. However branch 1.0 is no longer supported, that's why there wasn't patch for it. Could you please change PR target branch to 1.1 (and rebase)?

micszo commented:
@adamwojs @ITernovtsiy in fact it could be rebased to 1.2 since 1.1 is past EOM also (https://support.ibexa.co/Public/Service-Life), unless this is a customer issue I suppose.

adamwojs commented:
You are right @micszo. My bad.

ITernovtsiy commented:
> in fact it could be rebased to 1.2 since 1.1 is past EOM also (https://support.ibexa.co/Public/Service-Life),

Great, rebased to `1.2` )

> unless this is a customer issue I suppose.

I have extended and fixed this service in project code, so it's not an issue for the customer anymore)

Thanks.

bogusez commented:
@ITernovtsiy should your change fix the issue for content created before fix?

ITernovtsiy commented:
@bogusez No, and I don't think it's possible, as you'll not be able to determine where content sorting was intentionally changed to Name/Asc, and where it was a result of the bug.
Probably, we can develop some simple command which will accept content type and change all locations to match default sorting (where location is Name/Asc).

bogusez commented:
> @bogusez No, and I don't think it's possible, as you'll not be able to determine where content sorting was intentionally changed to Name/Asc, and where it was a result of the bug.
> Probably, we can develop some simple command which will accept content type and change all locations to match default sorting (where location is Name/Asc).

Ok thank you, just wanted to make sure.

adamwojs commented:
Thank you for contribution @ITernovtsiy!
